### ISSUE TITLE
Crash on save after pasting phone number into iPhoneNumberField

### ISSUE THREAD

**Issue by plorimer** [marmelroy/PhoneNumberKit#713]
### New Issue Checklist

- [x] Updated PhoneNumberKit to the latest version
- [x] Phone number formatted correctly on [JavaScript version](https://htmlpreview.github.io/?https://github.com/google/libphonenumber/blob/master/javascript/i18n/phonenumbers/demo-compiled.html)
- [x] I searched for [existing GitHub issues](https://github.com/marmelroy/PhoneNumberKit)

### Steps to reproduce
When a user pastes a phone number into an iPhoneNumberField on a form, then saves the object, the app immediatly crashes. If the number is typed into the field it saves correctly. I am seeing this on an iPad Pro 11" running under iPadOS 16.6.1, yet the same code runs without error on an iPhone 14 Pro Max running under iOS 17.0.3

This is the code that's in my form:

```
                        GridRow {
                            Text("Phone:")
                                .font(.headline)
                            iPhoneNumberField("Phone", text: $phone)
                                .flagHidden(false)
                                .flagSelectable(true)
                                .clearButtonMode(.whileEditing)
                                .formatted(true)
                                .padding(.horizontal, 10)
                                .padding(.vertical, 5)
                                .background(Color(UIColor.systemBackground))
                                .cornerRadius(5)
                        }

```

Not sure if this is an iPhoneNumberField (0.10.1) error or PhoneNumberKit (3.7.4) error

##### Expected result
Expect form to save into Core Data

##### Actual result
app crashes

### Environment
iOS integrated into Xcode via packages

**Comment by plorimer** [comment]
I found the cause. Not really iPhoneNumberField's problem.
When long clicking on the phone number field in Safari it gives the option to "copy phone number" Apparently this brought is white space. Then when I displayed the phone number as a clickable link:  ` Link(phone, destination: URL(string: "tel:\(phone)")!)` the unwrapping caught the whitespace and crashed, reporting it as unwrapping a `nil` value.

My solution was to add the following when saving the phone number into Core Data:  `person.phone = phone.filter { !$0.isWhitespace }`

Is there any reason that iPhoneNumberFiled couldn't strip whitespace internally, or is that required by some countries?
